bug: reNgine Installation failed: "sudo ./install.sh -n" | "unknown shorthand flag: 'f' in -f" | make: *** [Makefile:17: certs] Error 125

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

After cloning the rengine repo and changing the "POSTGRES_PASSWORD" password and giving the execution permissions with the command: "chmod +x install.sh". I have executed the following command: sudo ./install.sh -n, but I got an error at "unknown shorthand flag: 'f' in -f" and "make: *** [Makefile:17: certs] Error 125".

I have attached the screenshots for reference, please kindly look into it and let me know the possible resolution so that I can work on it to fix the issue from my end.

Reference Screenshots:

![rengine 001](https://github.com/yogeshojha/rengine/assets/16838353/cd47d1a2-ce07-49a7-ad4f-b6bfecfe266b)
![rengine 002](https://github.com/yogeshojha/rengine/assets/16838353/1a43a979-b932-44dd-a693-2df645858ccb)
![rengine 003](https://github.com/yogeshojha/rengine/assets/16838353/5304f2a5-0131-4f56-a48d-e99e20f1affd)


### Expected Behavior

It needs to be executed without any issues.

### Steps To Reproduce

Followed the required steps:
1. git clone https://github.com/yogeshojha/rengine && cd rengine
2. nano .env
3. chmod +x install.sh
4. sudo ./install.sh or sudo ./install.sh -n

### Environment

```markdown
- reNgine: 2.1.0
- OS: Linux kali 6.8.11-amd64
- Python: 3.11.9
- Docker Engine: 
- Docker Compose: 20.10.25+dfsg1
```


### Anything else?

_No response_

--- a/Makefile
+++ b/Makefile
@@ -9,59 +9,63 @@ COMPOSE_PREFIX_CMD := COMPOSE_DOCKER_CLI_BUILD=1
 COMPOSE_ALL_FILES := -f docker-compose.yml
 SERVICES          := db web proxy redis celery celery-beat ollama
 
+# Check if 'docker compose' command is available, otherwise use 'docker-compose'
+DOCKER_COMPOSE := $(shell if command -v docker > /dev/null && docker compose version > /dev/null 2>&1; then echo "docker compose"; else echo "docker-compose"; fi)
+$(info Using: $(shell echo "$(DOCKER_COMPOSE)"))
+
 # --------------------------
 
 .PHONY: setup certs up build username pull down stop restart rm logs
 
 certs:		    ## Generate certificates.
-	@${COMPOSE_PREFIX_CMD} docker compose -f docker-compose.setup.yml run --rm certs
+	@${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} -f docker-compose.setup.yml run --rm certs
 
 setup:			## Generate certificates.
 	@make certs
 
 up:				## Build and start all services.
-	${COMPOSE_PREFIX_CMD} docker compose ${COMPOSE_ALL_FILES} up -d --build ${SERVICES}
+	${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} ${COMPOSE_ALL_FILES} up -d --build ${SERVICES}
 
 build:			## Build all services.
-	${COMPOSE_PREFIX_CMD} docker compose ${COMPOSE_ALL_FILES} build ${SERVICES}
+	${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} ${COMPOSE_ALL_FILES} build ${SERVICES}
 
 username:		## Generate Username (Use only after make up).
 ifeq ($(isNonInteractive), true)
-	${COMPOSE_PREFIX_CMD} docker compose ${COMPOSE_ALL_FILES} exec web python3 manage.py createsuperuser --username ${DJANGO_SUPERUSER_USERNAME} --email ${DJANGO_SUPERUSER_EMAIL} --noinput
+	${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} ${COMPOSE_ALL_FILES} exec web python3 manage.py createsuperuser --username ${DJANGO_SUPERUSER_USERNAME} --email ${DJANGO_SUPERUSER_EMAIL} --noinput
 else
-	${COMPOSE_PREFIX_CMD} docker compose ${COMPOSE_ALL_FILES} exec web python3 manage.py createsuperuser
+	${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} ${COMPOSE_ALL_FILES} exec web python3 manage.py createsuperuser
 endif
 
 changepassword:	## Change password for user
-	${COMPOSE_PREFIX_CMD} docker compose ${COMPOSE_ALL_FILES} exec web python3 manage.py changepassword
+	${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} ${COMPOSE_ALL_FILES} exec web python3 manage.py changepassword
 
 migrate:		## Apply migrations
-	${COMPOSE_PREFIX_CMD} docker compose ${COMPOSE_ALL_FILES} exec web python3 manage.py migrate
+	${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} ${COMPOSE_ALL_FILES} exec web python3 manage.py migrate
 
 pull:			## Pull Docker images.
 	docker login docker.pkg.github.com
-	${COMPOSE_PREFIX_CMD} docker compose ${COMPOSE_ALL_FILES} pull
+	${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} ${COMPOSE_ALL_FILES} pull
 
 down:			## Down all services.
-	${COMPOSE_PREFIX_CMD} docker compose ${COMPOSE_ALL_FILES} down
+	${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} ${COMPOSE_ALL_FILES} down
 
 stop:			## Stop all services.
-	${COMPOSE_PREFIX_CMD} docker compose ${COMPOSE_ALL_FILES} stop ${SERVICES}
+	${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} ${COMPOSE_ALL_FILES} stop ${SERVICES}
 
 restart:		## Restart all services.
-	${COMPOSE_PREFIX_CMD} docker compose ${COMPOSE_ALL_FILES} restart ${SERVICES}
+	${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} ${COMPOSE_ALL_FILES} restart ${SERVICES}
 
 rm:				## Remove all services containers.
-	${COMPOSE_PREFIX_CMD} docker compose $(COMPOSE_ALL_FILES) rm -f ${SERVICES}
+	${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} $(COMPOSE_ALL_FILES) rm -f ${SERVICES}
 
 test:
-	${COMPOSE_PREFIX_CMD} docker compose $(COMPOSE_ALL_FILES) exec celery python3 -m unittest tests/test_scan.py
+	${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} $(COMPOSE_ALL_FILES) exec celery python3 -m unittest tests/test_scan.py
 
 logs:			## Tail all logs with -n 1000.
-	${COMPOSE_PREFIX_CMD} docker compose $(COMPOSE_ALL_FILES) logs --follow --tail=1000 ${SERVICES}
+	${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} $(COMPOSE_ALL_FILES) logs --follow --tail=1000 ${SERVICES}
 
 images:			## Show all Docker images.
-	${COMPOSE_PREFIX_CMD} docker compose $(COMPOSE_ALL_FILES) images ${SERVICES}
+	${COMPOSE_PREFIX_CMD} ${DOCKER_COMPOSE} $(COMPOSE_ALL_FILES) images ${SERVICES}
 
 prune:			## Remove containers and delete volume data.
 	@make stop && make rm && docker volume prune -f

--- a/make.bat
+++ b/make.bat
@@ -6,12 +6,13 @@ set COMPOSE_ALL_FILES  = -f docker-compose.yml
 set SERVICES           = db web proxy redis celery celery-beat ollama
 
 :: Check if 'docker compose' command is available
-where docker-compose >nul 2>&1
+docker compose version >nul 2>&1
 if %errorlevel% == 0 (
-    set DOCKER_COMPOSE=docker-compose
-) else (
     set DOCKER_COMPOSE=docker compose
+) else (
+    set DOCKER_COMPOSE=docker-compose
 )
+
 
 :: Generate certificates.
 if "%1" == "certs" %DOCKER_COMPOSE% -f docker-compose.setup.yml run --rm certs

--- a/make.bat
+++ b/make.bat
@@ -3,35 +3,43 @@
 :: Credits: https://github.com/ninjhacks
 
 set COMPOSE_ALL_FILES  = -f docker-compose.yml
-set SERVICES           = db web proxy redis celery celery-beat
+set SERVICES           = db web proxy redis celery celery-beat ollama
+
+:: Check if 'docker compose' command is available
+where docker-compose >nul 2>&1
+if %errorlevel% == 0 (
+    set DOCKER_COMPOSE=docker-compose
+) else (
+    set DOCKER_COMPOSE=docker compose
+)
 
 :: Generate certificates.
-if "%1" == "certs" docker compose -f docker-compose.setup.yml run --rm certs
+if "%1" == "certs" %DOCKER_COMPOSE% -f docker-compose.setup.yml run --rm certs
 :: Generate certificates.
-if "%1" == "setup" docker compose -f docker-compose.setup.yml run --rm certs
+if "%1" == "setup" %DOCKER_COMPOSE% -f docker-compose.setup.yml run --rm certs
 :: Build and start all services.
-if "%1" == "up" docker compose %COMPOSE_ALL_FILES% up -d --build %SERVICES%
+if "%1" == "up" %DOCKER_COMPOSE% %COMPOSE_ALL_FILES% up -d --build %SERVICES%
 :: Build all services.
-if "%1" == "build" docker compose %COMPOSE_ALL_FILES% build %SERVICES%
+if "%1" == "build" %DOCKER_COMPOSE% %COMPOSE_ALL_FILES% build %SERVICES%
 :: Generate Username (Use only after make up).
-if "%1" == "username" docker-compose %COMPOSE_ALL_FILES% exec web python3 manage.py createsuperuser
+if "%1" == "username" %DOCKER_COMPOSE% %COMPOSE_ALL_FILES% exec web python3 manage.py createsuperuser
 :: Apply migrations
-if "%1" == "migrate" docker-compose %COMPOSE_ALL_FILES% exec web python3 manage.py migrate
+if "%1" == "migrate" %DOCKER_COMPOSE% %COMPOSE_ALL_FILES% exec web python3 manage.py migrate
 :: Pull Docker images.
-if "%1" == "pull" docker login docker.pkg.github.com & docker-compose %COMPOSE_ALL_FILES% pull
+if "%1" == "pull" %DOCKER_COMPOSE% docker.pkg.github.com & docker-compose %COMPOSE_ALL_FILES% pull
 :: Down all services.
-if "%1" == "down" docker compose %COMPOSE_ALL_FILES% down
+if "%1" == "down" %DOCKER_COMPOSE% %COMPOSE_ALL_FILES% down
 :: Stop all services.
-if "%1" == "stop" docker compose %COMPOSE_ALL_FILES% stop %SERVICES%
+if "%1" == "stop" %DOCKER_COMPOSE% %COMPOSE_ALL_FILES% stop %SERVICES%
 :: Restart all services.
-if "%1" == "restart" docker compose %COMPOSE_ALL_FILES% restart %SERVICES%
+if "%1" == "restart" %DOCKER_COMPOSE% %COMPOSE_ALL_FILES% restart %SERVICES%
 :: Remove all services containers.
-if "%1" == "rm" docker compose %COMPOSE_ALL_FILES% rm -f %SERVICES%
+if "%1" == "rm" %DOCKER_COMPOSE% %COMPOSE_ALL_FILES% rm -f %SERVICES%
 :: Tail all logs with -n 1000.
-if "%1" == "logs" docker compose %COMPOSE_ALL_FILES% logs --follow --tail=1000 %SERVICES%
+if "%1" == "logs" %DOCKER_COMPOSE% %COMPOSE_ALL_FILES% logs --follow --tail=1000 %SERVICES%
 :: Show all Docker images.
-if "%1" == "images" docker compose %COMPOSE_ALL_FILES% images %SERVICES%
+if "%1" == "images" %DOCKER_COMPOSE% %COMPOSE_ALL_FILES% images %SERVICES%
 :: Remove containers and delete volume data.
-if "%1" == "prune" docker compose %COMPOSE_ALL_FILES% stop %SERVICES% & docker-compose %COMPOSE_ALL_FILES% rm -f %SERVICES% & docker volume prune -f
+if "%1" == "prune" %DOCKER_COMPOSE% %COMPOSE_ALL_FILES% stop %SERVICES% & docker-compose %COMPOSE_ALL_FILES% rm -f %SERVICES% & docker volume prune -f
 :: Show this help.
 if "%1" == "help" @echo Make application Docker images and manage containers using Docker Compose files only for windows.

